Изменен диалог создания/правки математической формулы

DikBSD opened the pull request:
Изменен диалог создания/правки математической формулы.

1. На диалоге создано 2 области:
- область создания/правки формулы (внизу диалога)
- область просмотра картинки формулы (вверху диалога)
2. Области разделены перемещающимся вертикально разделителем.
3. Область создания/правки формулы можно масштабировать кнопками Zoom In и Zoom Out.
4. В диалоге "Настройки редактора" создан контрол для установки периода обновления картинки формулы по таймеру, в секундах.
5. Обновление картинки формулы происходит либо в режиме установленного в диалоге "Настройки редактора" периода для таймера (задано по-умолчанию), либо - в режиме заданного реального времени.
6. Чекбокс "Fit to scroll area" позволяет либо масштабировать картинку в пределах области прокрутки картинки (включено), либо - масштабирование отключено, и картинка "растет" вправо по мере создания формулы.
7. При перемещении разделителя, если чекбокс "Fit to scroll area" включен, то происходит масштабирование картинки в границах области прокрутки картинки.
8. Все временные картинки формулы не переносятся в папку trash, а полностью удаляются. Для этого были внесены изменения в метод createGifFromMathExpression класса MathExpressionFormatter. По-умолчанию метод MathExpressionFormatter работает, как обычно - переносит промежуточные данные по картинкам в папку trash в момент вставки формулы. Но при правке формулы временные картинки и Tex файлы удаляются полностью.
9. Сделано обновление картинки формулы и для undo/redo редактора формул.
10. Минимальные размеры диалога заданы относительно от размеров приложения MyTetra.

DikBSD committed:
Изменен диалог создания/правки математической формулы.

1. На диалоге создано 2 области:
- область создания/правки формулы (внизу диалога)
- область просмотра картинки формулы (вверху диалога)
2. Области разделены перемещающимся вертикально разделителем.
3. Область создания/правки формулы можно масштабировать кнопками Zoom In
и Zoom Out.
4. В диалоге "Настройки редактора" создан контрол для установки периода
обновления картинки формулы по таймеру, в секундах.
5. Обновление картинки формулы происходит либо в режиме установленного в
диалоге "Настройки редактора" периода для таймера (задано по-умолчанию),
либо - в режиме заданного реального времени.
6. Чекбокс "Fit to scroll area" позволяет либо масштабировать картинку в
пределах области прокрутки картинки (включено), либо - масштабирование
отключено, и картинка "растет" вправо по мере создания формулы.
7. При перемещении разделителя, если чекбокс "Fit to scroll area"
включен, то происходит масштабирование картинки в границах области
прокрутки картинки.
8. Все временные картинки формулы не переносятся в папку trash, а
полностью удаляются. Для этого были внесены изменения в метод
createGifFromMathExpression класса MathExpressionFormatter. По-умолчанию
метод MathExpressionFormatter работает, как обычно - переносит
промежуточные данные по картинкам в папку trash в момент вставки
формулы. Но при правке формулы временные картинки и Tex файлы удаляются
полностью.
9. Сделано обновление картинки формулы и для undo/redo редактора формул.
10. Минимальные размеры диалога заданы относительно от размеров
приложения MyTetra.

xintrea commented:
Благодарю, Вадим. Базис заложен, но вот какие замечания:

1. При установке галки Fit... обновление картинки происходит, а при снятии - нет. И картинка остается растянутой на всю область.

2. Галка Fit... должна влиять на картинку только, если картинка больше по высоте или ширине области отрисовки формулы. Иначе у тебя маленькие картинки растягиваются на весь размер, и это выглядит жутко.

3. Серый фон в незанятой формулой области редактирования слишком темный. Там нужен цет light... или вручную задать что-то в светло-голубой спектор, что-то типа d7eef4.

4. В настройках редактора я попытался поставить на обновление таймера 2 секунды вместо 1 (нажимал экранную кнопочку со стрелкой вверх) и получил сегфолт.

DikBSD commented:
Сергей, по поводу 4 пункта  -это, по-видимому, мой косяк в диалоге распределения кнопок по панелям инструментов. Там с диалога берется флаг, были ли изменения, или нет. А потом по циклу уже в Вашем коде прогоняются все панели диалога настроек. По-видимому, если диалог работы с кнопками не был вызван, этот злополучный флаг не был создан в памяти, и в Вашем цикле происходит обращение "в никуда". И - как следствие - падение программы.
Я не учел, что диалог работы с кнопками может и не вызываться, а флаг на предмет изменения расположения кнопок на панелях будет вызываться в любом случае...

Давайте я исправлю это падение в новом реквесте, а потом займусь Вашими замечаниями, хорошо?

xintrea commented:
Каку удобней будет, так и сделайте

DikBSD commented:
Хорошо.
Если успею сегодня - исправлю баг и вышлю
Если не успею - то после выходных.

xintrea commented:
Ок

DikBSD commented:
Сергей СЮДА, пишу, чтобы не создавать ветку с сообщениями...
Если не ошибаюсь, то перестали загружаться картинки по копи/пасте с сайтов: Либо диалог с прогрессами копирования картинок висит, либо - вставляется просто контур картинки 1x1 см. размером.
Возможно, это произошло после того, как Вы делали преобразование html из скопированного с сайта текста в QTextDocument, а потом все это помещали в редактор. Возможно...

DikBSD committed:
Сделано изменение цвета фона области прокрутки картинки формулы с Dark на Light

DikBSD commented:
> При установке галки Fit... обновление картинки происходит, а при снятии - нет. И картинка остается растянутой на всю область.
Не смог воспроизвести. Изменение картинки происходит и при включенной галке, и при отключенной. При отключенной галке картинка уменьшается при удалении символов из формулы.

xintrea commented:
А этот баг воспроизводится на http сайтах, или только на https?

Если на https, то прочитайте вот эту тему: https://www.linux.org.ru/forum/development/14233348

Возможно у вас в сиетеме просто не те библиотеки openssl.

xintrea commented:
> При установке галки Fit... обновление картинки происходит, а при снятии - нет. И картинка остается растянутой на всю область.
> Не смог воспроизвести. Изменение картинки происходит и при включенной галке, и при отключенной. При отключенной галке картинка уменьшается при удалении символов из формулы.

Вот. Тут нарушено основное правило юзабилити: вы делаете действие (ставите галку) и ожидаете увидеть результат. Но  результата нет, он появится при последующих действиях. Это неправильно. Там очень сложно дёрнуть обновление картинки и при выставлении галки и при снятии?

DikBSD commented:
> А этот баг воспроизводится на http сайтах, или только на https?
> 
> Если на https, то прочитайте вот эту тему: https://www.linux.org.ru/forum/development/14233348
> 
> Возможно у вас в сиетеме просто не те библиотеки openssl.

Не смог на скорую руку найти сайт с http://

Вот из этого сайта - картинка не вставляется, только контур: 
http://www.cyberforum.ru/csharp-beginners/thread2031051.html

DikBSD commented:
> Там  очень сложно дёрнуть обновление картинки и при выставлении галки и при снятии?

Сделал утром, поэтому и удалил это свое сообщение.

Сейчас "мучаюсь" над 

> Галка Fit... должна влиять на картинку только, если картинка больше по высоте или ширине области отрисовки формулы. Иначе у тебя маленькие картинки растягиваются на весь размер, и это выглядит жутко.

Скоро пришлю, что получилось (немного не нравится, доделываю), посмотрите тогда, все ли так...

xintrea commented:
> Вот из этого сайта - картинка не вставляется, только контур:
> http://www.cyberforum.ru/csharp-beginners/thread2031051.html

Проверил. Картинка с логотипом Cuberforum загружается нормально. А вот картинка с кодом - нет. Почему? Наверно потому что они там накрутили какую-то защиту, может быть по referer. Если вы возьмете URL картинки, и попробуете ее напрямую открыть в другом браузере, например вот эту:

http://www.cyberforum.ru/attachments/861044d1503608800t

то у вас ничего не получится, сервер вам ее не отдаст. Чрезе wget тоже картинку невозмоджно получить:

> $ wget http://www.cyberforum.ru/attachments/861044d1503608800t
> --2019-03-11 16:33:36--  http://www.cyberforum.ru/attachments/861044d1503608800t
> Распознаётся www.cyberforum.ru (www.cyberforum.ru)… 185.112.80.14
> Подключение к www.cyberforum.ru (www.cyberforum.ru)|185.112.80.14|:80... соединение установлено.
> HTTP-запрос отправлен. Ожидание ответа… 403 Forbidden
> 2019-03-11 16:33:36 ОШИБКА 403: Forbidden.

Downloader MyTeta с точки зрения сайта cyberforum - это тоже просто какой-то сторонний браузер, который пытается открыть картинку, не зайдя на саму страницу (ведь мы на страницу не заходим через MyTetra, а делаем копипаст текста в браузере, и потом берем со страницы через MyTetra только картинки).

DikBSD commented:
Понятно.

DikBSD commented:
Сергей, как бы нам с Вами связаться по скайпу, вайберу или еще как... Хочу обсудить с Вами некоторые вопросы - минут на 10 от силы. Про развитие программы. Есть идеи, реализация которых выведет mytetra на новый уровень. Но нужно Ваше мнение по этим вопросам.
**Вот мой скайп: `kvesda`**
Если нужно, могу прислать номер телефона.

xintrea commented:
Скайп не хочет находить пользователя kvesda. Вы точно не ошиблись?

DikBSD commented:
Да, это мой скайп: kvesda

пн, 11 мар. 2019 г., 18:38 Xintrea <notifications@github.com>:

> Скайп не хочет находить пользователя kvesda. Вы точно не ошиблись?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/xintrea/mytetra_dev/pull/116#issuecomment-471592431>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeyE22Uz8MLmkEoJy9x3ZzWXwmm3gMgks5vVniPgaJpZM4birvl>
> .
>

xintrea commented:
То, что обещал: 

https://www.linux.org.ru/news/opensource/14069888

Qt5 Configuration Tool

DikBSD commented:
Спасибо. 
--
Отправлено из Mail.Ru для Android понедельник, 11 марта 2019г., 20:29 +03:00 от Xintrea  notifications@github.com :

>То, что обещал:
>https://www.linux.org.ru/news/opensource/14069888
>Qt5 Configuration Tool
>—
>You are receiving this because you authored the thread.
>Reply to this email directly,  view it on GitHub , or  mute the thread .

DikBSD committed:
Изменен алгоритм масштабирования картинки формулы.

1. Перемещение разделителя не влияет на масштабирование картинки.
2. При включенной опции "Fit to scroll area for big formula" картинка,
размеры которой превышают размеры области прокрутки, масштабируется.
3. При выключенной опции "Fit to scroll area for big formula" картинка
не масштабируется, а отображается, как есть.
4. При изменении размеров диалога создания/редактирования формулы
происходит масштабирование картинки формулы, согласно алгоритма опции
"Fit to scroll area for big formula".

DikBSD commented:
Сергей, давай на "ты", хорошо?
Я выслал исправленный код. Посмотри, пожалуйста, как теперь?
